javascript/testing/README.md

```diff
@@ -487,6 +491,65 @@ jest.mock('moment', ()=> ({
 }))
 ```
 
+## End-to-end Testing
+> For End-to-end testing we use [Cypress](https://www.cypress.io/) Framework. However, our suggested concepts could be adopted with other End-to-End frameworks/test runners. 
+
+The below diagram illustrates main blocks that involved in End-to-end test. 
+
+![End-to-end Test diagram](./resources/e2eDiagram.png)
+
+- **Test** 
+    - Uses the [AAA pattern](#aaa-pattern---arrange-act-assert)
+    - One and only one unit that __has assertions__ for testing components' integration and data integrity 
+    - Uses one or more `Page Objects` to access and interact with tested application page
+    
+Here is a simple example of test that checks search data functionality:    
+
+``` javascript
+import TestPageObject from './TestPageObject'
+const testPageObject =  new TestPageObject()
+
+it('should search data correctly', () => {
+  
+  testPageObject.typeSearchTerm('searchTerm')
+  
+  testPageObject.getSearchResults().should('contain', 'searchTerm')
+})
+```
+   
+- **Page Object** 
+   - Responsible to encapsulate technical details (like css selectors, data attributes, etc`) to access the elements of the tested application page
+   - Provides an atomic API for interaction with tested application page. This API is used by `Test` 
```

Comment:
"Provides an API for atomic interactions with the tested application page" - I think that the operation should be atomic and not API

Reply:
fixed